### ISSUE TITLE
`@param` parser fails when braces appear in description.

### ISSUE THREAD

**Issue by rhys-vdw** [esdoc/esdoc#185]
Just trying out esdoc for the first time. ~~I'm getting errors everywhere I'm using unions in parens.~~ This is supported syntax according to [docs](https://esdoc.org/tags.html#nullable-and-not-nullable).

```js
   * @param {?(Object|function)} initializer
   *  An initializer callback, taking the Options instance as its first
   *  argument. Alternatively an object of {[method]: argument} pairs to be
   *  invoked.
```

```console
/Users/rhys/Projects/atlas/data-mapper/node_modules/esdoc/out/src/Publisher/Builder/ClassDocBuilder.js:79
            throw _iteratorError;
                  ^
SyntaxError: Invalid regular expression: /[~]\function)} initializer
 An initializer callback, taking the Options instance as its first
 argument. Alternatively an object of {[method]: argument$/: Unmatched ')'
    at RegExp (native)
    at new RegExp (/Users/rhys/Projects/atlas/data-mapper/node_modules/esdoc/node_modules/core-js/modules/es6.regexp.js:22:76)
    at ClassDocBuilder._findByName (/Users/rhys/Projects/atlas/data-mapper/node_modules/esdoc/out/src/Publisher/Builder/DocBuilder.js:140:20)
    at ClassDocBuilder._buildDocLinkHTML (/Users/rhys/Projects/atlas/data-mapper/node_modules/esdoc/out/src/Publisher/Builder/DocBuilder.js:1032:22)
    at ClassDocBuilder._buildTypeDocLinkHTML (/Users/rhys/Projects/atlas/data-mapper/node_modules/esdoc/out/src/Publisher/Builder/DocBuilder.js:1008:21)
    at ClassDocBuilder._buildSignatureHTML (/Users/rhys/Projects/atlas/data-mapper/node_modules/esdoc/out/src/Publisher/Builder/DocBuilder.js:1141:33)
    at /Users/rhys/Projects/atlas/data-mapper/node_modules/esdoc/out/src/Publisher/Builder/DocBuilder.js:500:38
    at IceCap.loop (/Users/rhys/Projects/atlas/data-mapper/node_modules/ice-cap/out/src/IceCap.js:319:15)
    at ClassDocBuilder._buildSummaryDoc (/Users/rhys/Projects/atlas/data-mapper/node_modules/esdoc/out/src/Publisher/Builder/DocBuilder.js:497:11)
    at ClassDocBuilder._buildSummaryHTML (/Users/rhys/Projects/atlas/data-mapper/node_modules/esdoc/out/src/Publisher/Builder/DocBuilder.js:456:29)
```

```console
npm ls | grep esdoc
├─┬ esdoc@0.4.3
├─┬ esdoc-es7-plugin@0.0.3
```

**Comment by rhys-vdw** [comment]
Seems this was caused by having "{ ... }" in the description. If I wrap it in backticks it works as intended.

```js
   * @param {?(Object|function)} initializer
   *   An initializer callback, taking the Options instance as its first
   *   argument. Alternatively an object of `{[method]: argument}` pairs to be
   *   invoked.
```

**Comment by h13i32maru** [comment]
Thanks for reporting this issue. I fixed this issue on master.

**Comment by h13i32maru** [comment]
I released v0.4.5 that fixed this issue.
